File: src/main/java/com/refinedmods/refinedstorage/tile/WirelessTransmitterTile.java
package com.refinedmods.refinedstorage.tile;

import com.refinedmods.refinedstorage.RSTiles;
import com.refinedmods.refinedstorage.apiimpl.network.node.WirelessTransmitterNetworkNode;
import com.refinedmods.refinedstorage.tile.data.TileDataParameter;
import net.minecraft.core.BlockPos;
import net.minecraft.network.syncher.EntityDataSerializers;
import net.minecraft.world.level.Level;
import net.minecraft.world.level.block.state.BlockState;

import javax.annotation.Nonnull;

public class WirelessTransmitterTile extends NetworkNodeTile<WirelessTransmitterNetworkNode> {
    public static final TileDataParameter<Integer, WirelessTransmitterTile> RANGE = new TileDataParameter<>(EntityDataSerializers.INT, 0, t -> t.getNode().getRange());

    public WirelessTransmitterTile(BlockPos pos, BlockState state) {
        super(RSTiles.WIRELESS_TRANSMITTER, pos, state);

        dataManager.addWatchedParameter(RANGE);
    }

    @Override
    @Nonnull
    public WirelessTransmitterNetworkNode createNode(Level world, BlockPos pos) {
        return new WirelessTransmitterNetworkNode(world, pos);
    }
}

File: src/main/kotlin/com/refinedmods/refinedstorage/tile/WirelessTransmitterTile.kt
package com.refinedmods.refinedstorage.tile

import com.refinedmods.refinedstorage.RSTiles
import com.refinedmods.refinedstorage.apiimpl.network.node.WirelessTransmitterNetworkNode
import com.refinedmods.refinedstorage.tile.data.TileDataParameter
import net.minecraft.network.datasync.DataSerializers
import net.minecraft.util.math.BlockPos
import net.minecraft.world.World
import java.util.function.Function

class WirelessTransmitterTile : NetworkNodeTile<WirelessTransmitterNetworkNode?>(RSTiles.WIRELESS_TRANSMITTER) {
    @Nonnull
    override fun createNode(world: World?, pos: BlockPos?): WirelessTransmitterNetworkNode {
        return WirelessTransmitterNetworkNode(world, pos)
    }

    companion object {
        val RANGE = TileDataParameter(DataSerializers.VARINT, 0, Function { t: WirelessTransmitterTile -> t.getNode()!!.range })
    }

    init {
        dataManager.addWatchedParameter(RANGE)
    }
}